test/test.ads.js

```diff
@@ -729,6 +729,45 @@ QUnit.test('adsready in content-playback triggers readyforpreroll', function(ass
   assert.strictEqual(spy.getCall(0).args[0].type, 'readyforpreroll', 'readyforpreroll should have been triggered.');
 });
 
+QUnit.test('contentupdate in preroll? transitions to content-set', function(assert) {
+  assert.expect(2);
+
+  this.player.trigger('play');
+  this.player.trigger('adsready');
+  assert.strictEqual(this.player.ads.state, 'preroll?');
+  this.player.trigger('contentupdate');
+  assert.strictEqual(this.player.ads.state, 'content-set');
+});
+
+QUnit.test('contentupdate in content-playback transitions to content-set if the player is paused', function(assert) {
+  assert.expect(2);
+
+  this.player.trigger('play');
+  this.player.trigger('adtimeout');
```

Comment:
I would just do `this.players.ads.skipLinearAdMode()` here, or explicitly do `startLinearAdMode()` and  `endLinearAdMode()` right after.